pep-0708.rst

```diff
@@ -0,0 +1,849 @@
+work for pip isn't a particularly useful solution.*
+
+
+Implicitly allow mirrors when the list of files are the same
+------------------------------------------------------------
+
+If every repository returns the exact same list of files, then it is safe to
+consider those repositories to be the same namespace and implicitly merge them.
+This would possibly mean that mirrors would be automatically allowed without any
+work on any user or repository operator's part.
+
+Unfortunately, this has two failings that make it undesirable:
+
+- It only solves the case of mirrors that are exact copies of each other, but
+  not repositories that "track" another one, which ends up being a more generic
+  solution.
+- Even in the case of exact mirrors, multiple repositories mirroring each other
+  is a distributed system will not always be fully consistent with each
+  other, effectively an eventually consistent system. This means that
+  repositories that relied on this implicit heuristic to work would have
+  sporadic failures due to drift between the source repository and the mirror
+  repositories.
+
+
+Provide a mechanism to order the repositories
+---------------------------------------------
+
+Providing some mechanism to give the repositories an order, and then short
+circuiting the discovery algorithm when it finds the first repository that
+provides files for that project is another workable solution that is safe if the
+order is specified correctly.
+
+However, this has been rejected for a number of reasons:
+
+- We've spent 15+ years educating users that the ordering of repositories being
+  specified is not meaningful, and they effectively have an undefined order. It
+  would be difficult to backpedal on that and start saying that now order
+  matters.
+- Users can easily rearrange the order that they specify their repositories in
+  within a single location, but when loading repositories from multiple
+  locations (env var, conf file, requirements file, cli arguments) the order is
+  hard coded into pip. While it would be a deterministic and documented order,
+  there's no reason to assume it's the order that the user wants their
+  repositories to be defined in, forcing them to contort thow they configure pip
+  so that the implicit ordering ends up being the correct one.
+- The above can be mitigated by providing a way to explicitly declare the order
+  rather than by implicitly using the order they were defined in, however that
+  then means that the protections are not provided unless the user does some
+  explicit configuration.
+- Ordering assumes that there is a linear ordering of repositories where two
+  repositories have the same name, we always prefer repositories in the same
+  order for every project, however that is not necessarily true.
+- Relying on ordering is subtle, if I look at an ordering of repositories, I
+  have no way of knowing in advance or ensuring in advance what names are going
```

Comment:
```suggestion
  have no way of knowing or ensuring in advance what names are going
```

I've generally avoided suggesting fixes to repetition issues, but here it results in the syntax being particularly contorted, has a clear solution and seems likely to be an editing typo